OpenRA.Game/MiniYaml.cs

```diff
@@ -141,6 +141,34 @@ public MiniYaml WithNodesAppended(IEnumerable<MiniYamlNode> nodes)
 			return new MiniYaml(Value, newNodes);
 		}
 
+		public MiniYamlNode NodeWithKey(string key)
+		{
+			var result = NodeWithKeyOrDefault(key);
+			if (result == null)
+				throw new InvalidDataException($"No node with key '{key}'");
+			return result;
+		}
+
+		public MiniYamlNode NodeWithKeyOrDefault(string key)
```

Comment:
suggestion `FindChild(string key)`.

Reply:
I could go for `FindNode(string key)`?

Reply:
Although that would pair with `FindNodeOrDefault` since we want both to have two overloads, one allowing no key and one requiring a key.

Reply:
IMO the current naming is the clearest of all these, and is consistent with LINQ.

Reply:
Will keep as-is then.